src/Uno.UI/UI/Xaml/Controls/Layouter/Layouter.cs

```diff
@@ -97,20 +129,44 @@ public Size Measure(Size availableSize)
 					}
 				}
 
-				SetDesiredChildSize(this.Panel as View, desiredSize);
+				SetDesiredChildSize(Panel as View, desiredSize);
+
+				var clippedDesiredSize = desiredSize
+					.Add(marginSize)
+					.AtMost(availableSize)
+					.AtLeast(new Size(0, 0));
 
-				return size;
+				return clippedDesiredSize;
 			}
 		}
 
+		private static bool IsLessThanAndNotCloseTo(double a, double b)
+		{
+			return a < b - SIZE_EPSILON;
+
+			//if (IsPositiveInfinity(a))
```

Comment:
Remove ?

Reply:
Still WIP...  You know, it's a DRAFT - WIP PR will almost all commits named "WIP" ;-)